high-perms-checkout/action.yml

```diff
@@ -68,7 +68,7 @@ runs:
     # builders are run concurrently. This is simplified
     # for our e2e tests.
     - name: Upload the artifact
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
```

Comment:
Might we want to pin these to a version as well? example-package is in many ways more problematic security-wise than slsa-github-generator repo since example-package has access to highly privileged PAT tokens.

Reply:
this repo only contains example. Either we pin it and we'll never update, or we keep a floating version. I'm fine pinning so long as dependabot is _not_ enabled :)

Reply:
SG